src/LinSolvers.jl

```diff
@@ -40,29 +41,42 @@ module LinSolvers
     """
         The linear solver associated with julia factorize()
     """
-    struct DefaultLinSolver{T_num<:Number, T_mat<:AbstractMatrix} <: LinSolver
-        A::T_mat
+    struct DefaultLinSolver <: LinSolver
         Afact
-
+        umfpack_refinements::Int
     end
-    DefaultLinSolver(A::AbstractMatrix,Afact) = new(A, Afact)
 
-    function DefaultLinSolver(nep::NEP,λ)
+    function DefaultLinSolver(nep::NEP, λ, umfpack_refinements)
         A=compute_Mder(nep,λ)
         Afact=factorize(A)
-        return DefaultLinSolver{eltype(A),typeof(A)}(A, Afact)
+        return DefaultLinSolver(Afact, umfpack_refinements)
     end
```

Comment:
I also don't directly see the use of having parametric types here. It was turned parametric at a point before we fully understood how to use parametric types well (march) 

However, in #44 we made newton + DEP type stable. With these type of changes might loose that? To be sure not to loose this along the way, we could add a unit test corresponding verifying type stability similar to #44.

Reply:
I meant `augnewton` is type stable for DEPs: 
```
julia> dep=nep_gallery("dep0");
julia> @code_warntype(augnewton(Float64,dep))
```
So a short unit test checking could something like:
```
dep=nep_gallery("dep0");
Test.@inferred augnewton(Float64,dep,v=ones(size(dep,1)))
Test.@inferred augnewton(ComplexF64,dep,v=ones(size(dep,1)))

```

Reply:
Yes, we did not fully understand how to use parametric types. However, considering the following [performance tips](https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type-1) in the Julia manual it is possible that we still want to have the types parametric. Although maybe something like
```
    struct DefaultLinSolver{T} <: LinSolver
        Afact::T
        umfpack_refinements::Int
    end

    function DefaultLinSolver(nep::NEP, λ, umfpack_refinements))
        A = compute_Mder(nep,λ)
        Afact = factorize(A)
        return DefaultLinSolver{typeof(Afact)}(Afact, umfpack_refinements))
    end
```

Reply:
Good suggestions. I've updated the code accordingly.